### DISCUSSION TITLE
Say I have scrolled horizontally rightward how do I get the first visible line number?

### DISCUSSION THREAD

**Discussion by nyngwang** [neovim/neovim#24033] [Q&A] [answered]
As title.

My trying: I found that `vim.fn.virtcol('.')` can give what I want **only if** I place my cursor at the first visible column after, e.g. `z10l`, in the first place. Now the problem becomes what should I pass to `virtcol`.

**Comment by luukvbaal** [comment, marked as answer]
`vim.fn.winsaveview()` contains the `leftcol` field, that should be what you're looking for.

**Reply by nyngwang** [reply]
Thank you!
